foreign-key.md

```diff
@@ -0,0 +1,312 @@
+
+The effect of disabling [`foreign_key_checks`](/system-variables.md#foreign_key_checks) is as follows:
+
+- When you delete a parent table referenced by a foreign key, the deletion can succeed only when the foreign key constraint check is disabled.
+- When you import data to a database, the order of creating tables might be different from the foreign key dependency order, which might cause the creation of tables to fail. Only when the foreign key constraint check is disabled can the tables be created successfully. In addition, disabling the foreign key constraint check can speed up data import.
+- When you import data to a database, if the data of the child table is imported first, an error will be reported. Only when the foreign key constraint check is disabled can the data of the child table be imported successfully.
+- If an `ALTER TABLE` operation to be executed involves a change of the foreign key, this operation succeeds only when the foreign key constraint check is disabled.
+
+When the foreign key constraint check is disabled, the foreign key constraint check and reference operation are not executed, except for the following scenarios:
+
+- If the execution of `ALTER TABLE` might result in wrong definition of the foreign key, an error is still reported during the execution.
+- When deleting the index required by the foreign key, you should delete the foreign key first. Otherwise, an error is reported.
+- When you create a foreign key but it does not meet related conditions or restrictions, an error is reported.
+
+## Locking
+
+When `INSERT` or `UPDATE` a child table, the foreign key constraint checks whether the corresponding foreign key value exists in the parent table, and locks the row in the parent table to avoid the foreign key value being deleted by other operations violating the foreign key constraint. The locking behavior is equivalent to performing a `SELECT FOR UPDATE` operation on the row where the foreign key value is located in the parent table.
+
+Because TiDB currently does not support `LOCK IN SHARE MODE`, if there are a lot of concurrent writes to the child table and most of the referenced foreign key values are the same, there might be serious locking conflicts. It is recommended to disable [`foreign_key_checks`](/system-variables.md#foreign_key_checks) when writing a large number of child table data.
+
+## Definition and metadata of foreign keys
+
+To view the definition of a foreign key constraint, use the `SHOW CREATE TABLE` statement:
+
+```sql
+mysql> SHOW CREATE TABLE child\G
+*************************** 1. row ***************************
+       Table: child
+Create Table: CREATE TABLE `child` (
+  `id` int(11) DEFAULT NULL,
+  `pid` int(11) DEFAULT NULL,
+  KEY `idx_pid` (`pid`),
+  CONSTRAINT `fk_1` FOREIGN KEY (`pid`) REFERENCES `test`.`parent` (`id`) ON DELETE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+```
+
```

Comment:
```suggestion

You can also get information about foreign keys using either of the following system table:

- [`INFORMATION_SCHEMA.KEY_COLUMN_USAGE`](/information-schema/information-schema-key-column-usage.md)
- [`INFORMATION_SCHEMA.TABLE_CONSTRAINTS`](/information-schema/information-schema-table-constraints.md)
- [`INFORMATION_SCHEMA.TABLE_CONSTRAINTS`](/information-schema/information-schema-table-constraints.md)

The following provides examples:

```

Reply:
@shichun-0415 The docs-cn PR is updated: https://github.com/pingcap/docs-cn/pull/11974/commits/52cbed4cd9c05805798c6074ebf5d9492bb26922